client/tests/integration/upgrade.rs

```diff
@@ -225,7 +225,45 @@ fn migration_fail_should_not_cause_any_effects() {
     // been changed, because `executor_with_migration_fail` does not allow any queries
 }
 
-fn upgrade_executor(client: &Client, executor: impl AsRef<Path>) -> Result<()> {
+// TODO: check that upgrade event contains data model
+
+#[test]
+fn migration_should_cause_upgrade_event() {
+    let (_rt, _peer, client) = <PeerBuilder>::new().with_port(10_996).start_with_runtime();
+    wait_for_genesis_committed(&vec![client.clone()], 0);
+
+    let (sender, receiver) = mpsc::channel();
+    let events_client = client.clone();
+    let _handle = thread::spawn(move || {
+        for event in events_client
+            .listen_for_events([ExecutorEventFilter::new()])
+            .unwrap()
+            .map(Result::unwrap)
+        {
+            if let EventBox::Data(DataEvent::Executor(ExecutorEvent::Upgraded(ExecutorUpgrade {
+                new_data_model,
+            }))) = event
+            {
+                let _ = sender.send(new_data_model);
+            }
+        }
+    });
```

Comment:
Instead of spawning threads we can use async `listen_for_events_async` and `rt.block_on`.
But since this is a test i don't think it's critical here.

Reply:
Thanks for notice, refactored with `async`.